### PR TITLE
Docs: Updated API - ref for payments

### LINKED ISSUE #5014: Updating the API reference documentation
We have had feedback related to the API doc being outdated, and lacking proper visibility of the latest spec of the API collection.

### LINKED ISSUE #5174: Updating the API reference documentation - payments
We have had feedback related to the API doc being outdated, and lacking proper visibility of the latest spec of the API collection.

### PR DESCRIPTION
**Pull request by gorakhnathy7**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [X] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Partially closes #5174 

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -911,7 +912,7 @@ impl Default for MandateType {
     }
 }
 
-/// We will be Passing this "CustomerAcceptance" object during Payments-Confirm. The customer_acceptance sub object is usually passed by the SDK or client
+/// This "CustomerAcceptance" object during Payments-Confirm. The customer_acceptance sub object is usually passed by the SDK or client
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:915]: Can we improve this?

File: crates/api_models/src/payments.rs [hunk 2]
```diff
@@ -4250,7 +4251,7 @@ pub struct ApplepaySessionRequest {
     pub initiative_context: String,
 }
 
-/// additional data related to some connectors
+/// Some connectors like Apple pay, Airwallex and Noon might require some additional information, find specific details in the child attributes below.
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:4254]:
```suggestion
/// Some connectors like Apple Pay, Airwallex and Noon might require some additional information, find specific details in the child attributes below.
```

File: crates/api_models/src/payments.rs [hunk 3]
```diff
@@ -298,7 +299,7 @@ pub struct PaymentsRequest {
     #[remove_in(PaymentsUpdateRequest, PaymentsCreateRequest, PaymentsConfirmRequest)]
     pub capture_on: Option<PrimitiveDateTime>,
 
-    /// Whether to confirm the payment (if applicable)
+    /// Whether to confirm the payment (if applicable). It can be used to completely process a payment by attaching a payment method, setting `confirm=true` and `capture_method = automatic` in the *Payments/Create request itself.
```
[the remark this answers is not in the document]
Comment (gorakhnathy7) [reply crates/api_models/src/payments.rs:302]: corrected

File: crates/api_models/src/payments.rs (hunk 1 above)
Comment (gorakhnathy7) [reply crates/api_models/src/payments.rs:915]: Done

**Review by SanchithHegde — APPROVED** [review]

File: crates/openapi/src/routes/payments.rs [hunk 4]
```diff
@@ -1,10 +1,12 @@
 /// Payments - Create
 ///
-/// **Creates a payment object when amount and currency are passed.** This API is also used to create a mandate by passing the `mandate_object`.
+/// **Creates a payment object when amount and currency are passed.**
 ///
-/// To completely process a payment you will have to create a payment, attach a payment method, confirm and capture funds.
```
Comment (Narayanbhat166) [inline crates/openapi/src/routes/payments.rs:5]: why is this removed?

**Review by Narayanbhat166 — APPROVED** [review]
